Softeq.XToolkit.Common.Tests/Tasks/TaskDeferralTests.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using System.Threading.Tasks;
+using Softeq.XToolkit.Common.Tasks;
+using Xunit;
+
+namespace Softeq.XToolkit.Common.Tests.Tasks
+{
+    public class TaskDeferralTests
+    {
+        private readonly TaskDeferral<string> _deferral;
+
+        public TaskDeferralTests()
+        {
+            _deferral = new TaskDeferral<string>();
+        }
+
+        [Fact]
+        public async Task DoWorkAsync_Null_ReturnsNull()
+        {
+            var result = await _deferral.DoWorkAsync(null!);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task DoWorkAsync_OneTaskWithResult_ReturnsResult()
+        {
+            var funcTask = CreateTaskWithResult("test", 10);
```

Comment:
I've already tried it, seems like in these cases better to keep tests solid (because moving string/int into separate variables make it difficult to read a difficult test)

Reply:
Okay)